devices/Mpu6886/Mpu6886.cs

```diff
@@ -0,0 +1,419 @@
+        /// Gets and sets the gyroscope offset in the GyroscopeOffsetAdjustment registers of the MPU6886.
+        /// Setting the offset can be usefull when a custom callibration calculation is used, instead of the Calibrate function of this class.
+        /// </summary>
+        public Vector3 GyroscopeOffset
+        {
+            get
+            {
+                SpanByte vec = new byte[6];
+                Read(Mpu6886.Register.GyroscopeOffsetAdjustmentXHighByte, vec);
+
+                Vector3 v = new Vector3();
+                v.X = (short)(vec[0] << 8 | vec[1]);
+                v.Y = (short)(vec[2] << 8 | vec[3]);
+                v.Z = (short)(vec[4] << 8 | vec[5]);
+
+                return v;
+            }
+
+            set
+            {
+                SpanByte registerAndOffset = new byte[7];
+                SpanByte offsetbyte = new byte[2];
+
+                registerAndOffset[0] = (byte)Mpu6886.Register.GyroscopeOffsetAdjustmentXHighByte;
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.X);
+                registerAndOffset[1] = offsetbyte[0];
+                registerAndOffset[2] = offsetbyte[1];
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.Y);
+                registerAndOffset[3] = offsetbyte[0];
+                registerAndOffset[4] = offsetbyte[1];
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.Z);
+                registerAndOffset[5] = offsetbyte[0];
+                registerAndOffset[6] = offsetbyte[1];
+
+                _i2c.Write(registerAndOffset);
+            }
+        }
+
+        /// <summary>
+        /// Reset the internal registers and restores the default settings. (Datasheet, page 47)
+        /// </summary>
+        public void Reset()
+        {
+            _i2c.Write(new SpanByte(new byte[] { (byte)Mpu6886.Register.PowerManagement1, 0b1000_0000 }));
+            Thread.Sleep(10);
+        }
+
+        /// <summary>
+        /// Set the chip to sleep mode. (Datasheet, page 47)
+        /// </summary>
+        public void Sleep()
+        {
+            _i2c.Write(new SpanByte(new byte[] { (byte)Mpu6886.Register.PowerManagement1, 0b0100_0000 }));
+            Thread.Sleep(10);
+        }
+
+        /// <summary>
+        /// Disables the sleep mode. (Datasheet, page 47)
+        /// </summary>
+        public void WakeUp()
+        {
+            _i2c.Write(new SpanByte(new byte[] { (byte)Mpu6886.Register.PowerManagement1, 0b0000_0000 }));
+            Thread.Sleep(10);
+        }
+
+        /// <summary>
+        /// Gets and sets the accelerometer full scale. (Datasheet page 37)
+        /// </summary>
+        public AccelerometerScale AccelerometerScale
+        {
+            get
+            {
+                SpanByte buffer = new byte[1];
+                Read(Mpu6886.Register.AccelerometerConfiguration1, buffer);
+                return (AccelerometerScale)buffer[0];
```

Comment:
Are you sure it's the only element in this register? In all cases, better to adjust like:
```
return (AccelerometerScale)(buffer[0] & 0b0001_1000);
```

Reply:
It's only the 3 bits to start the self test which are in the register (together with the scale). So potentially when the self test would be implemented, this could become an issue. Easy enough to avoid, so adjusting now.